File: anika.dovob/src/main/java/com/example/anika/homework/chapter7/Order.java
package com.example.anika.homework.chapter7;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity(name = "Ch7Order")
@Table(name = "orders_ch7")
public class Order extends BaseEntity {

    @Id
    @GeneratedValue
    @Column(name = "order_id")
    private Long id;

    @Column(name = "order_no", unique = true, nullable = false)
    private String orderNo;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> orderItems = new ArrayList<>();

    protected Order() {
    }

    public Order(String orderNo) {
        this.orderNo = orderNo;
    }

    public void addOrderItem(OrderItem orderItem) {
        orderItems.add(orderItem);
        orderItem.setOrder(this);
    }

    public Long getId() {
        return id;
    }

    public String getOrderNo() {
        return orderNo;
    }

    public List<OrderItem> getOrderItems() {
        return orderItems;
    }
}

File: hyun.k/hyun.k/src/main/kotlin/chapter7/Order.kt
package chapter7

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity
@Table(name = "orders")
class Order() : BaseEntity() {
    @Id
    @GeneratedValue
    var id: Long? = null

    @Column(unique = true, nullable = false)
    var orderNo: String = ""

    // TODO: OrderItem 연관관계
    @OneToMany(mappedBy = "order")
    var orderItems = mutableListOf<OrderItem>()
}